jbpm-persistence/jbpm-persistence-jpa/src/main/java/org/jbpm/persistence/JpaProcessPersistenceContext.java

```diff
@@ -66,7 +67,10 @@ public PersistentProcessInstance findProcessInstanceInfo(Long processId) {
     	if( this.pessimisticLocking ) { 
             return em.find( ProcessInstanceInfo.class, processId, lockMode );
         }
-        return em.find( ProcessInstanceInfo.class, processId );
+        logger.trace("Reading process instance info {} with em {}", processId, em);
+        ProcessInstanceInfo pi = em.find(ProcessInstanceInfo.class, processId);
+        logger.trace("Process instance info read {}", pi);
```

Comment:
To avoid computational time when trace level is not enabled -basically for the find operation-, you could include them inside an if clause, wdyt?:
```suggestion
        if (logger.isTraceEnabled()) {
          logger.trace("Reading process instance info {} with em {}", processId, em);
          ProcessInstanceInfo pi = em.find(ProcessInstanceInfo.class, processId);
          logger.trace("Process instance info read {}", pi);
      }
```

Reply:
this will disable the find